src/main/java/io/strimzi/kafka/bridge/mqtt/core/MqttServerHandler.java

```diff
@@ -91,15 +110,76 @@ private void handleConnectMessage(ChannelHandlerContext ctx) {
         ctx.writeAndFlush(connAckMessage);
     }
 
+    /**
+     * Send a MQTT PUBACK message to the client.
+     *
+     * @param ctx ChannelHandlerContext instance
+     * @param packetId packet identifier
+     */
+    private void sendPubAckMessage(ChannelHandlerContext ctx, int packetId) {
+        MqttMessage pubAckMessage = MqttMessageBuilders.pubAck()
+                .packetId(packetId)
+                .build();
+
+        ctx.writeAndFlush(pubAckMessage);
+    }
+
     /**
      * Handle the case when a client sent a MQTT PUBLISH message type.
      *
      * @param ctx            ChannelHandlerContext instance
      * @param publishMessage represents a MqttPublishMessage
      */
     private void handlePublishMessage(ChannelHandlerContext ctx, MqttPublishMessage publishMessage) {
-        logger.info("MQTT Topic: {}", publishMessage.variableHeader().topicName());
-        logger.info("Kafka Mapped Topic: {}", mqttKafkaMapper.map(publishMessage.variableHeader().topicName()));
-        logger.info("Message: {}", publishMessage.payload().toString(Charset.defaultCharset()));
+        // get QoS level from the MqttPublishMessage
+        MqttQoS qos = MqttQoS.valueOf(publishMessage.fixedHeader().qosLevel().value());
+
+        // perform topic mapping
+        String mappedTopic = mqttKafkaMapper.map(publishMessage.variableHeader().topicName());
+
+        // build the Kafka record
+        ProducerRecord<String, ByteBuf> record = new ProducerRecord<>(mappedTopic,
+                publishMessage.payload());
```

Comment:
We are just moving bytes from MQTT payload to Kafka payload, but it doesn't allow to the MQTT client to specify a key (see above comment).
Usually the MQTT payload is something more than just bytes, i.e. a JSON that could contains key and value as we have for the Strimzi HTTP bridge. The JSON could also bring more than one message/record.
As starting point I am fine with moving bytes and not having a key, but we should track with an issue and think about it as next step.

Reply:
I have opened #20 to keep track of this.